runtime/major_gc.c

```diff
@@ -584,7 +584,7 @@ static void adopt_orphaned_work (int expected_status)
   orph_ephe_list_verify_status(expected_status);
 #endif
 
-  if (no_orphaned_work() || caml_domain_is_terminating())
```

Comment:
Does this change mean (a) a terminating domain may adopt orphaned work or (b) a terminating domain will not even call `adopt_orphaned_work` i.e., `assert(!caml_domain_is_terminating())` will not hold? Which one is it?

Reply:
Reading further, I understand it is (a).